.github/workflows/build_and_test.yml

```diff
@@ -58,7 +74,7 @@ jobs:
     container:
       image: ubuntu:bionic
     needs:
-    - build_and_test_master
+    - build_and_test
```

Comment:
Previously we only cared to report metrics for the builds on release-latest. This will make it so that we now care for both master as well as release-latest. Is that intended?

If so, can you not instead do
```
matrix:
        ros_distro: [kinetic, melodic, dashing]
        ros_distro: [kinetic, melodic, dashing]
        vcs-repo-file-url: ["", "${{ github.workspace }}/.github/workflows/release_latest.repos"]
```

And then merge the code from 47-63

Reply:
That won't work given the variable expansion, where the variable would be empty until the checkout action is used.

Reply:
Both master and release latest builds were previously reported. There's a bug here where only the master branch is currently set.

Reply:
Sorry, I shouldnt be doing reviews at midnight. 

I confused myself, both master and release-latest were reported but using separate tags to cloudwatch. With this change, we are consolidating (or trying to) both reporting just one metric of failure or success for both master and release-latest.

Reply:
No it's a good point: I'm not sure how to differentiate between the failures given the consolidation. It might be good enough to say: this workflow failed for some reason and have the person dig into which action actually triggered the failure.